Primo.js

```diff
@@ -278,32 +287,87 @@ function importPNX(text) {
 	
 	fetchCreators(item, creators, 'author', splitGuidance);
 	fetchCreators(item, contributors, 'contributor', splitGuidance);
+
+	// For this section, contrastive examples:
+	// http://primo.bib.uni-mannheim.de/primo_library/libweb/action/display.do?doc=MAN_ALEPH000967205&showPnx=true
+	// http://digitale.beic.it/primo_library/libweb/action/display.do?doc=39bei_digitool2018516&showPnx=true
+	// http://digitale.beic.it/primo_library/libweb/action/display.do?doc=39bei_digitool2394625&showPnx=true
+	// Example: "Milano". Rarely: "Mediolani", "In Milano" etc.
+	// BEIC.it also uses //display/lds09, //search/lsr12, //search/lsr03, //facets/lfc03
+	item.place = ZU.xpathText(doc, '//addata/cop');
 	
+	// E.g. "Parma : Viotto", "Ambrogio : da Caponago", "Mantegazza, Filippo"
 	var publisher = ZU.xpathText(doc, '//display/publisher');
-	if(publisher) var pubplace = ZU.unescapeHTML(publisher).split(" : ");
-	if(pubplace && pubplace[1]) {
-		item.place = pubplace[0].replace(/,\s*c?\d+|[\(\)\[\]]|(\.\s*)?/g, "");
+	// E.g. "Viotto", "[Ambrogio da Caponago]", "per magistro Philippo dicto Cassano di Mantegatii"
+	if(!publisher) var publisher = ZU.xpathText(doc, '//addata/pub');
+	if(publisher) {
+		var pubplace = ZU.unescapeHTML(publisher).split(" : ");
+		var possibleplace = pubplace[0].replace(/,\s*c?\d+|[\(\)\[\]]|(\.\s*)?/g, "");
+	}
+	if(pubplace && pubplace[1] && item.place && item.place !== possibleplace) {
+		// The splitting produced something, but the first segment is not the place.
+		// Probably, the whole string was the publisher's name.
+		// Should also catch cases like "Meurs, Jacob : van "
+		item.publisher = publisher;
+	} else if(pubplace && pubplace[1]) {
+		// Hopefully we split in place, publisher name.
 		item.publisher = pubplace[1].replace(/,\s*c?\d+|[\(\)\[\]]|(\.\s*)?/g, "")
 			.replace(/^\s*"|,?"\s*$/g, '');
 	} else if(pubplace) {
-		item.publisher = pubplace[0].replace(/,\s*c?\d+|[\(\)\[\]]|(\.\s*)?/g, "")
-			.replace(/^\s*"|,?"\s*$/g, '');
+		// No splitting, use the whole (cleaned) string
+		item.publisher = possibleplace.replace(/^\s*"|,?"\s*$/g, '');
 	}
-	
+	if(!item.place && possibleplace && pubplace[1]) {
+		// Splitting happened and we need a place, we hope this is it
+		item.place = possibleplace;
+	}
```

Comment:
The different cases here are very hard to read. Maybe, we can make that cases easier, i.e. something like
```
item.place = ZU.xpathText(doc, '//addata/cop');
var publisher = ZU.xpathText(doc, '//display/publisher');
if(!publisher) publisher = ZU.xpathText(doc, '//addata/pub');
if(publisher) {
	publisher = publisher.replace(/,\s*c?\d+|[\(\)\[\]]|(\.\s*)?/g, "");
	item.publisher = publisher.replace(/^\s*"|,?"\s*$/g, '');
	var pubplace = ZU.unescapeHTML(publisher).split(" : ");
	
	if(pubplace && pubplace[1]) {
		var possibleplace = pubplace[0];
		if(!item.place ) {
			item.publisher = pubplace[1].replace(/^\s*"|,?"\s*$/g, '');
			item.place = possibleplace;
		}
		if(item.place && item.place == possibleplace) {
			item.publisher = pubplace[1].replace(/^\s*"|,?"\s*$/g, '');
		}
	}
}
```
Please not that I haven't tested this code and you might want to add some comments again.